### ISSUE TITLE
Different array order of values each execution when preloading struct

### ISSUE THREAD

**Issue by shackra** [go-gorm/gorm#2868]
Your issue may already be reported! Please search on the [issue track](https://github.com/jinzhu/gorm/issues) before creating one.

### What version of Go are you using (`go version`)?
1.13.x

### Which database and its version are you using?
postgres 12.1

### Please provide a complete runnable program to reproduce your issue. **IMPORTANT**

Maybe we won't need one.

### Issue

I have the following ORM query:

```
// ...
var documento model.Documento
	err := db.Preload(
		"Grupo").Preload(
		"Receptor").Preload(
		"Emisor").Preload(
		"OtrosCargos").Preload(
		"LineaDetalle").Preload(
		"LineaDetalle.Impuesto").Where(
		"clave = ?", c).First(
		&documento).Error
// ...
```

And I noticed that in every execution one SQL query has the values of an array in different order

```
[2020-01-29 21:35:59]  [0.22ms]  SELECT * FROM "impuestos"  WHERE "impuestos"."deleted_at" IS NULL AND (("linea_detalle_id" IN (10,1,8,9,5,6,7,2,3,4))) ORDER BY "impuestos"."id" ASC  
[0 rows affected or returned ] 

[2020-01-29 21:32:13]  [0.14ms]  SELECT * FROM "impuestos"  WHERE "impuestos"."deleted_at" IS NULL AND (("linea_detalle_id" IN (3,5,6,9,1,4,7,8,10,2))) ORDER BY "impuestos"."id" ASC  
[0 rows affected or returned ] 

(/home/jorge/code/kue/fero/dev/backend/utils.go:450) 
[2020-01-29 21:23:49]  [0.24ms]  SELECT * FROM "impuestos"  WHERE "impuestos"."deleted_at" IS NULL AND (("linea_detalle_id" IN (7,10,1,2,3,6,4,5,8,9))) ORDER BY "impuestos"."id" ASC  
[0 rows affected or returned ] 
```

The issue with this is that affects mocking with `go-sqlmock`, I have an automatic process that retrieves data from a test database and spits everything needed for mocking a test and for the queries I'm just using `regexp.QuoteMeta` so the queries always match, but with this behavior from GORM is not feasible.

hopefully, this behavior is a bug. I don't think this relates to #598, #793 or #824

**Comment by jamie-34254** [comment]
We're also encountering this issue but we're not using the preload functionality

**Comment by shackra** [comment]
> We're also encountering this issue but we're not using the preload functionality
> 
> Looks like this might be fixed by this PR: #2880 ?

I work around this for my use case by switching to go-mocket
